cmd/collector/app/sanitizer/zipkin/span_sanitizer.go

```diff
@@ -83,8 +83,29 @@ type spanDurationSanitizer struct {
 func (s *spanDurationSanitizer) Sanitize(span *zc.Span) *zc.Span {
 	if span.Duration == nil {
 		span.Duration = &defaultDuration
+		if len(span.Annotations) < 2 {
+			return span
+		}
+		// Prefer RPC one-way (cs -> sr) vs arbitrary annotations.
+		first := span.Annotations[0].Timestamp
+		last := span.Annotations[len(span.Annotations) - 1].Timestamp
+		for _, anno := range span.Annotations {
+			if anno.Value == zc.CLIENT_SEND {
+				first = anno.Timestamp
+			} else if anno.Value == zc.CLIENT_RECV {
+				last = anno.Timestamp
+			}
+		}
+		if first != last {
+			duration := last - first
+			span.Duration = &duration
```

Comment:
double allocation with L85 - better to declare local var `duration` and do `span.Duration = &duration` at the end of `if`